sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/DeploymentDependentResource.java

```diff
@@ -5,11 +5,14 @@
 import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
 import io.javaoperatorsdk.operator.ReconcilerUtils;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Creator;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Updater;
 import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
 import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependentResource;
 
 @KubernetesDependent(labelSelector = "app.kubernetes.io/managed-by=tomcat-operator")
-public class DeploymentDependentResource extends KubernetesDependentResource<Deployment, Tomcat> {
+public class DeploymentDependentResource extends KubernetesDependentResource<Deployment, Tomcat>
+    implements Creator<Deployment, Tomcat>, Updater<Deployment, Tomcat> {
```

Comment:
Cannot this be done by the init?

Reply:
No, the whole point is to declare what your dependent is doing explicitly. `init` is only there to set the defaults for subclasses of `AbstractDependentResource`.

Reply:
Hmm, won't in most cases actually a full CRUD resource? Should we have an adaptor for that?